orchagent/sfloworch.cpp

```diff
@@ -0,0 +1,700 @@
+
+    if (sai_rc != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to set session %lx on port %lx",
+                       session.m_sample_id, port_id);
+        return false;
+    }
+    return true;
+}
+
+bool SflowOrch::sflowDelPort(SflowSession &session, sai_object_id_t port_id)
+{
+    sai_attribute_t attr;
+    sai_status_t    sai_rc;
+
+    if (!sflowStatus)
+    {
+        return true;
+    }
+
+    attr.id = SAI_PORT_ATTR_INGRESS_SAMPLEPACKET_ENABLE;
+    attr.value.oid = SAI_NULL_OBJECT_ID;
+
+    sai_rc = sai_port_api->set_port_attribute(port_id, &attr);
+
+    if (sai_rc != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to delete session %lx on port %lx",
+                       session.m_sample_id, port_id);
+        return false;
+    }
+    return true;
+}
+
+bool SflowOrch::sflowGlobalConfigure(bool enable)
+{
+    bool ret = true;
+
+    for(auto& pair: gPortsOrch->getAllPorts())
+    {
+        auto& port = pair.second;
+        if (port.m_type != Port::PHY)
+        {
+            continue;
+        }
+
+        auto sflowInfo = m_sflowPortSessionMap.find(port.m_port_id);
+        if (sflowInfo == m_sflowPortSessionMap.end())
+        {
+            if (!enable)
+            {
+                continue;
+            }
+            SflowSession session;
+            uint32_t     rate = 0;
+
+            sflowGetDefaultSampleRate(port, rate);
+            session.rate = rate;
+            session.globalConfigured = true;
+            session.adminState = true;
+            auto e_sampleObj = m_sflowRateSampleMap.find(rate);
+
+            if (e_sampleObj != m_sflowRateSampleMap.end())
+            {
+                session.m_sample_id = e_sampleObj->second;
+            }
+            else 
+            {
+                if (!sflowCreateSession(session))
+                {
+                    SWSS_LOG_NOTICE("Creating sflow session with rate %d failed", rate);
+                    ret = false;
+                    continue;
+                }
+                m_sflowRateSampleMap[rate] = session.m_sample_id;
+                m_sflowSampleRefMap[session.m_sample_id] = 0;
+            }
+
+            if (sflowAddPort(session, port.m_port_id))
+            {
+                m_sflowPortSessionMap[port.m_port_id] = session;
+                m_sflowSampleRefMap[session.m_sample_id]++;
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Failed to add port %s to global session",
+                               port.m_alias.c_str());
+                ret = false;
+
+                if (m_sflowSampleRefMap[session.m_sample_id] == 0)
+                {
+                    if (sflowDestroySession(session))
+                    {
+                        m_sflowSampleRefMap.erase(session.m_sample_id);
+                        m_sflowRateSampleMap.erase(rate);
+                    }
+                }
+            }
+        }
+        else
```

Comment:
The if and else is long that its hard to find which is the matching if. Can you add a one line comment here?